Bottom: 532d0db03bd0865707322b49512e1dff8cd88d42
Top:    e3fadda321fa6d59db7a182edb2e2c8ed99c7e98
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:30:45 +0800

autofs-5.0.7 - add map entry null check in do_expire_direct()

Since we've seen a mount failure for this map entry it should exist
but add a null check in case it's been removed while we waited on
the lock.


---

diff --git a/daemon/direct.c b/daemon/direct.c
index 399ad0a..f85e1b5 100644
--- a/daemon/direct.c
+++ b/daemon/direct.c
@@ -1019,7 +1019,8 @@ static void *do_expire_direct(void *arg)
 		struct mapent *me;
 		cache_writelock(mt.mc);
 		me = cache_lookup_distinct(mt.mc, mt.name);
-		me->ioctlfd = -1;
+		if (me)
+			me->ioctlfd = -1;
 		cache_unlock(mt.mc);
 		ops->send_ready(ap->logopt, mt.ioctlfd, mt.wait_queue_token);
 		ops->close(ap->logopt, mt.ioctlfd);
